pkg/core/consensus/reduction/reduction_test.go

```diff
@@ -61,10 +61,15 @@ func TestCorrectHeader(t *testing.T) {
 		t.FailNow()
 	}
 
-	// We discard the message in the middle, as it will be an Agreement message,
-	// as a result of the emptying of the queue resulting on quorum.
-	<-gossipChan
-	r2 := <-gossipChan
+	// There is no guaranteed order between the agreement message, and the second reduction
+	// message. So, we attempt to get the right one before moving on.
+	var r2 message.Message
+	for {
+		r2 = <-gossipChan
+		if r2.Category() == topics.Reduction {
+			break
+		}
+	}
```

Comment:
Is this the most elegant way to do this? It seems to me that either the `gossipChan` should be drained, or closed (and the attempt to write to a closed channel be handled properly)

Reply:
I don't know if that is such a good idea, since we might drain or close the channel before the send has happened. Hence why I have the `for` in place.